qt/python/mantidqt/widgets/manageuserdirectories.py

```diff
@@ -12,18 +12,23 @@
 from mantidqt.utils.qt import import_qt
 
 
-ManageUserDirectories_cpp = import_qt('.._common', 'mantidqt.widgets', 'ManageUserDirectories')
+ManageUserDirectories = import_qt('.._common', 'mantidqt.widgets', 'ManageUserDirectories')
 
+manage_user_directories = None
 
-class ManageUserDirectories(ManageUserDirectories_cpp):
-    """
-    Small wrapper class around the Manage User Directories Window that
-    hides the help button which can lead to a crash.
-    See https://github.com/mantidproject/mantid/issues/26404.
 
-    This is a safe, temporary fix, for release 4.0.1.
-    """
+def open_mud(parent):
+    # Checks whether a Manage User Directories window is already open before opening another one.
+    global manage_user_directories
```

Comment:
Do we need this to be global? 
Wrapping this in a class and using a static method we can keep this tightly encapsulated. That was the class responsible for raising an existing one then defer opening a new one into this presenter. I.e.

```
class ManageUserDirectories(object):

# Note no self to ensure it's a class var
currently_open_mud = None

def open_mud(parent):
    if self.currently_open_mud:
       self.currently_open_mud.raise_()
    else:
        # This was the rest of the code that was if it was None
        self.currently_open_mud.open_mud(self)

def close_mud():
    self.currently_open_mud=None
```

Reply:
I see what you're saying but I'm not sure where I would create the instance of this class so that open_mud can be called from all of the different places where you can open MUD?

Reply:
So because it's a class variable (rather than an instance) the moment anyone imports the class Python will handle it for us

(Some details are eluded to here: https://rushter.com/blog/python-class-internals/ :
How Python stores class attributes )